examples/ConfigurationStoreBackup/BackupAppConfigurationStore.cs

```diff
@@ -0,0 +1,163 @@
+    {
+        private static readonly ConfigurationClient primaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("PrimaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly ConfigurationClient secondaryAppConfigClient = new ConfigurationClient(new Uri(Environment.GetEnvironmentVariable("SecondaryStoreEndpoint")), new ManagedIdentityCredential());
+        private static readonly QueueClient queueClient = new QueueClient(Environment.GetEnvironmentVariable("AzureWebJobsStorage"), "eventgridqueue");
+        private const int maxMessagesToRead = 32;
+
+        [FunctionName("BackupAppConfigurationStore")]
+        public static void Run([TimerTrigger("0 */10 * * * *")]TimerInfo myTimer, ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            if (primaryAppConfigClient == null || secondaryAppConfigClient == null)
+            {
+                log.LogError($"Could not connect to App Configuration Store. Please validate the store endpoints that you have provided.");
+                return;
+            }
+
+            // Block the thread so that another function instance cannot be triggered while this is still processing.
+            ProcessStorageQueueMessagesAsync(log).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        private static async Task ProcessStorageQueueMessagesAsync(ILogger log)
+        {
+            // Peek to see if there are events in the queue.
+            while ((await queueClient.PeekMessagesAsync().ConfigureAwait(false)).Value.Length > 0)
+            {
+                Response<QueueMessage[]> retrievedMessages = await queueClient.ReceiveMessagesAsync(maxMessagesToRead).ConfigureAwait(false);
+
+                // Extract list of key+labels from event data.
+                HashSet<KeyLabel> updatedKeyLabels = ExtractKeyLabelsFromEvents(retrievedMessages.Value, log);
+
+                // If there are any valid App Configuration events, update secondary store.
+                if (updatedKeyLabels.Count > 0)
+                {
+                    bool isBackupSuccessful = await BackupKeyValuesAsync(updatedKeyLabels, log).ConfigureAwait(false);
+                    if (!isBackupSuccessful)
+                    {
+                        // Abort this function without deleting retrievedMessages from storage queue.
+                        log.LogError($"Backup failed because of error(s) returned by Azure App Configuration service.");
+                        return;
+                    }
+                }
+
+                // Delete this batch of events from storage queue.
+                foreach (QueueMessage message in retrievedMessages.Value)
+                {
+                    log.LogDebug($"Deleting message: {message.MessageId}");
+                    await queueClient.DeleteMessageAsync(message.MessageId, message.PopReceipt).ConfigureAwait(false);
```

Comment:
Any transient error this may throw? We shouldn't crash the Function app for those.

Reply:
This may throw transient errors like - Pop receipt mismatch, or message not found. We can ignore those and continue processing.

---
In reply to: [412707275](https://github.com/Azure/AppConfiguration/pull/311#discussion_r412707275) [](ancestors = 412707275)